### PR TITLE
Remove dependency on the scikit-learn confusion matrix

### PR DESCRIPTION
**Pull request by mohitcek**
- I have added a method (`binary_confusion_matrix`) to the parent class (Metric).
- There are multiple print statements to compare the results from scikit-learn cofusion_matrix function and self.binary_confusion_matrix method. Please look at the example notebook, I will remove the print statements and scikit-learn dependency after the review.

### CODE REVIEW

File: examples/evaluations/classification/classification_metrics_demo.ipynb [hunk 1]
```diff
@@ -85,7 +85,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 2,
+   "execution_count": 3,
```
Comment (dylanbouchard) [inline examples/evaluations/classification/classification_metrics_demo.ipynb:88]: In the section `1. Introduction`, can you please ensure that it matches what's already in develop? Should be something like this:

Large language models (LLMs) used in classification use cases should be assessed for group fairness (if applicable). Similar to traditional person-level classification challenges in machine learning, these use cases present the risk of allocational harms. LangFair offers the following classification fairness metrics from the LLM fairness literature:

Predicted Prevalence Rate Disparity ([Feldman et al., 2015](https://arxiv.org/abs/1412.3756); [Bellamy et al., 2018](https://arxiv.org/abs/1810.01943); [Saleiro et al., 2019](https://arxiv.org/abs/1811.05577))
False Negative Rate Disparity ([Bellamy et al., 2018](https://arxiv.org/abs/1810.01943); [Saleiro et al., 2019](https://arxiv.org/abs/1811.05577))
False Omission Rate Disparity ([Bellamy et al., 2018](https://arxiv.org/abs/1810.01943); [Saleiro et al., 2019](https://arxiv.org/abs/1811.05577))
False Positive Rate Disparity ([Bellamy et al., 2018](https://arxiv.org/abs/1810.01943); [Saleiro et al., 2019](https://arxiv.org/abs/1811.05577))
False Discovery Rate Disparity ([Bellamy et al., 2018](https://arxiv.org/abs/1810.01943); [Saleiro et al., 2019](https://arxiv.org/abs/1811.05577))

File: examples/evaluations/classification/classification_metrics_demo.ipynb [hunk 2]
```diff
@@ -153,23 +153,61 @@
...
     "tags": []
    },
    "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-------------\n",
+      "FalseNegativeRateParity\n",
+      "Sklearn Confusion matrix, CM1:  [[2519 1032]\n",
```
Comment (dylanbouchard) [inline examples/evaluations/classification/classification_metrics_demo.ipynb:168]: Can we re-run this after removing the printing of the confusion matrix?

File: langfair/metrics/classification/metrics/baseclass/metrics.py [hunk 3]
```diff
@@ -36,3 +36,19 @@ def evaluate(
         Abstract method that needs to be implemented by the user when creating a new metric function.
         """
         pass
+
+    @staticmethod
```
Comment (dylanbouchard) [inline langfair/metrics/classification/metrics/baseclass/metrics.py:40]: Looks great!

File: langfair/metrics/classification/metrics/false_discovery.py [hunk 4]
```diff
@@ -74,10 +74,16 @@ def evaluate(
         cm1 = confusion_matrix(
             y_true[groups == unique_groups[0]], y_pred[groups == unique_groups[0]]
         )
+        print("-------------")
```
Comment (dylanbouchard) [inline langfair/metrics/classification/metrics/false_discovery.py:77]: Can we please remove these print statements or create a verbose option to not show them and set default to not print? Same comment for all child classes. Thanks!

**Review by dylanbouchard — CHANGES_REQUESTED** [review]
This PR looks great! Very minor changes requested: 
- Removing printing or set as optional (default no print)
- Update classification demo notebook to be consistent with what's currently in develop (it appears changes were not fetched)
- Please also fix the links in the other notebooks if you don't mind. 

Thanks Mohit!

File: langfair/metrics/classification/metrics/false_discovery.py (hunk 4 above)
Comment (mohitcek) [reply langfair/metrics/classification/metrics/false_discovery.py]: Removed the print statements

File: examples/evaluations/classification/classification_metrics_demo.ipynb (hunk 2 above)
Comment (mohitcek) [reply examples/evaluations/classification/classification_metrics_demo.ipynb]: Done

File: examples/evaluations/classification/classification_metrics_demo.ipynb (hunk 1 above)
Comment (mohitcek) [reply examples/evaluations/classification/classification_metrics_demo.ipynb]: I have updated the notebook manually, but I am trying to pull changes from develop branch.

**Review by dylanbouchard — APPROVED** [review]

**Pull request merged**